builtin/mainmenu/settings/components.lua

```diff
@@ -306,18 +306,31 @@ function make.flags(setting)
 				"label[0,0.1;" .. get_label(setting) .. "]",
 			}
 
-			local value = core.settings:get(setting.name) or setting.default
 			self.resettable = core.settings:has(setting.name)
 
 			checkboxes = {}
-			for _, name in ipairs(value:split(",")) do
-				name = name:trim()
-				if name:sub(1, 2) == "no" then
-					checkboxes[name:sub(3)] = false
-				elseif name ~= "" then
-					checkboxes[name] = true
+			for _, name in ipairs(setting.possible) do
+				checkboxes[name] = false
+			end
+			local function apply_flags(flag_string)
+				for _, name in ipairs(flag_string:split(",")) do
+					name = name:trim()
+					if name:sub(1, 2) == "no" then
```

Comment:
I've had this in MCL¹ and it's problematic when you have to flag named `nodes_something`. Need to check the table of known valid flags here.

¹: there's a lack of abbrevations to use for Minetest Classic

Reply:
I had assumed such flags weren't a thing judging by the previous code.
I'll implement the proper solution.

BTW, I would abbreviate Minetest Classic as MTC.

Reply:
I have a flag called `node_waving` and it's actually missing from the settings tab.